autorest/client.go

```diff
@@ -72,6 +74,52 @@ const (
 `
 )
 
+type logLevelType int
+
+const (
+	logLevelNone logLevelType = iota
+	logLevelBasic
+	logLevelFull
```

Comment:
If it were me, I'd emulate popular tools like `github.com/sirupsen/logrus` and [loggly](https://www.loggly.com/blog/node-js-libraries-make-sophisticated-logging-simpler/) and have logging levels something like this, I think the difference in expected output is a lot easier to reason about than 'full' vs. 'basic':

- debug
- info
- warn
- error
- fatal
- panic

Reply:
I like this idea.  How about we rename ```logLevelBasic``` to ```info```?  I'm debating if we should just include the request/response bodies in that or keep it separate, perhaps under ```debug```.  I feel like if I want to log requests/responses I'd want them in full although I can see how they could be quite large.  Thoughts?

Reply:
Or we put all of this under ```debug```?

Reply:
I like the way you're thinking. I'd keep bodies under `debug` to save space, and the fact that a request, response was seen under `info` unless you encounter some reason why that really doesn't make sense.

Reply:
Agreed.  I'm tempted to include headers under ```debug``` as well.  This way if I just want to see request/response times I don't have to mentally tease out the header info I don't care about.

Reply:
We should also follow the logging semantics defined in the [pipeline](https://github.com/Azure/azure-pipeline-go/blob/master/pipeline/core.go) so that in vNext it cleanly translates.